gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/jsr223/ScriptEngineCacheTest.java

```diff
@@ -35,6 +37,9 @@ public void shouldBeUtilityClass() throws Exception {
 
     @Test
     public void shouldGetEngineFromCache() {
+        String javaVersion = System.getProperty("java.version");
```

Comment:
is there another way to test this that perhaps doesn't require us to use nashorn? that way we could avoid binding to a particular jdk version/type.

Reply:
There doesn't seem to be any default ScriptEngines that ship with the JDK anymore. To actually test the current method as is we would probably need to add another ScriptEngineFactory that the service provider could load. There isn't an easy way to mock this and still test the actual implementation of the cache. I'm not sure if it is worth adding those other pieces to make the ScriptEngineManager load a ScriptEngine just to test this.